### DISCUSSION TITLE
I thought I want something simple :)

### DISCUSSION THREAD

**Discussion by AmosSzust** [community/community#35954] [New to GitHub] [answered]
Hi

I have a repo and I want to achieve the following:
Any Github user who finds my repo interesting and wants to contribute, will clone it locally, make a branch out of the master branch, do his magic, create a PR and send it to me for approval. I should be the **only person** who can approve a PR before merging it into the master branch, **even my own PRs**.

I set CODEOWNERS file with my nick name and set "Do not allow bypassing the above settings", the problem is that I can't approve my own PRs so I got stuck. This means that I have to **uncheck** the option "Do not allow bypassing the above settings" so I can bypass the approval step, so it looks like this option is problematic. Also, **who else will be able to bypass the approval step**?

As I had no choice and I unchecked the above option, other people can make their own changes, **only** by making a fork of the repo. Why is that? Why can't they just make a branch with "write access" to that branch alone and whenever they will make a PR, it will get to me for approval?

Thanks
Amos

**Comment by antonblr** [comment, marked as answer]
Do you see the following option/checkbox on own PRs (`merge without waiting for requirements`)?

<img width="700" alt="image" src="https://user-images.githubusercontent.com/44246099/195450473-f0dbf2b5-4c7e-4400-a85b-bfefd74d8e54.png">

It is regulated by this branch protection setting (should be unchecked):

<img width="700" alt="image" src="https://user-images.githubusercontent.com/44246099/195450765-d25f5b6d-55c4-40ea-8565-82cf56efdaaf.png">

**Reply by AmosSzust** [reply]
Hi @antonblr and thank you for replying.

I unchecked "Do not allow bypassing the above settings" so I have the red checkbox option, but I wonder who else will have this red checkbox? I don't want anyone except me to have it. Checking this option makes no sense if I can't approve my own PR as the owner of the repo (I checked this option "Require review from Code Owners" and added a CODEOWNERS file).

I unchecked "Do not allow bypassing the above settings" option only because otherwise I wasn't able to approve my own PRs.

One last thing: Can other people do changes to my repo only by forking? Is there a way to just let them create their own branch in my repo and create a PR for me to approve?

**Reply by antonblr** [reply]
>but I wonder who else will have this red checkbox? I don't want anyone except me to have it.

Only administrators / repo owners -> such permissions has to granted explicitly for your repo.

>Can other people do changes to my repo only by forking? Is there a way to just let them create their own branch in my repo and create a PR for me to approve?

Only by forking. Otherwise you will have to add them as repo Collaborators first.

**Reply by AmosSzust** [reply]
Thanks. 
If I understand you correctly, adding a user as a collaborator, will allow him to work with branches instead of forks but he will **not** have this red checkbox allowing him to bypass the approval step?

If so, how/where/to whom those permissions are added (administrators/repo owners)?

**Reply by antonblr** [reply]
https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-access-to-your-personal-repositories/inviting-collaborators-to-a-personal-repository

You dictate the access level.
